### PR TITLE
added RBAC role, binding and service account

### PR DESCRIPTION
**Pull request by leahnp**
To reflect changes in: https://github.com/samsung-cnct/k2-logging-fluentd-daemonset/pull/5

Also added RBAC for central fluentd to reflect: samsung-cnct/k2-logging-central-fluentd#1

### CODE REVIEW

File: central-fluentd/templates/fluentd.yaml [hunk 1]
```diff
@@ -1,14 +1,62 @@
+{{- if .Capabilities.APIVersions.Has "rbac.authorization.k8s.io/v1beta1"}}
```
Comment (coffeepac) [inline central-fluentd/templates/fluentd.yaml]: there is no 'end' statement for this, which I think is needed

File: central-fluentd/templates/fluentd.yaml [hunk 2]
```diff
@@ -1,14 +1,62 @@
...
   name: {{ .Values.name }}
+  namespace: {{ .Values.namespace }}
 spec:
   replicas: {{ .Values.replicas}}
   template:
     metadata:
       labels:
         app: {{ .Values.app }}
     spec:
+      serviceAccountName: {{ .Values.sa_name }}
```
Comment (coffeepac) [inline central-fluentd/templates/fluentd.yaml]: I believe that we always want to create the Deployment and it appears the service account is intended to only be created if RBAC is enabled.  I think the answer is:
- move the service account creation to after the missing 'end' block

File: central-fluentd/values.yaml [hunk 3]
```diff
@@ -1,6 +1,21 @@
-# Default values for central-fluentd.
+# Values for Service Account
+sa_name: fluentd-central
+sa_namespace: kube-system
```
Comment (coffeepac) [inline central-fluentd/values.yaml]: we've been putting this in kube-system but do we want to still?  I'm wondering if a logging namespace (kube-logging) might be more appropriate.

File: daemon-fluentd/templates/fluentd.yaml [hunk 4]
```diff
@@ -1,17 +1,69 @@
+{{- if .Capabilities.APIVersions.Has "rbac.authorization.k8s.io/v1beta1"}}
```
Comment (coffeepac) [inline daemon-fluentd/templates/fluentd.yaml]: same comments as above re: lacking end statement and what happens with the service account

File: central-fluentd/values.yaml (hunk 3 above)
Comment (leahnp) [inline central-fluentd/values.yaml]: yes, I think a logging specific namespace would be good, especially for central logging, which has lots of things to keep track of.

File: central-fluentd/templates/fluentd.yaml (hunk 1 above)
Comment (leahnp) [inline central-fluentd/templates/fluentd.yaml]: addressed

File: central-fluentd/templates/fluentd.yaml (hunk 2 above)
Comment (leahnp) [inline central-fluentd/templates/fluentd.yaml]: Addressed - let me know if that's what you were looking for

File: daemon-fluentd/templates/fluentd.yaml (hunk 4 above)
Comment (leahnp) [inline daemon-fluentd/templates/fluentd.yaml]: addressed

File: central-fluentd/templates/fluentd.yaml (hunk 2 above)
Comment (coffeepac) [inline central-fluentd/templates/fluentd.yaml]: yep.  thanks!

**Pull request merged**